File: app/libs/daypickerlibrary/src/main/java/com/github/informramiz/daypickerlibrary/common/CommonUtils.java
package com.github.informramiz.daypickerlibrary.common;

import android.content.Context;
import android.support.annotation.ColorRes;
import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.util.TypedValue;

/**
 * Created by ramiz on 2/1/18.
 */

public class CommonUtils {
    public static @ColorRes int resolveResource(@NonNull Context context, int resId) {
        final TypedValue outValue = new TypedValue();
        context.getTheme().resolveAttribute(resId, outValue, true);
        return outValue.resourceId;
    }
}

File: app/src/main/java/cn/starrah/thu_course_helper/picker/CommonUtils.kt
package cn.starrah.thu_course_helper.picker

import android.content.Context
import androidx.annotation.ColorRes
import androidx.annotation.IdRes
import android.util.TypedValue
import androidx.annotation.NonNull

/**
 * Created by ramiz on 2/1/18.
 */
object CommonUtils {
    @ColorRes
    fun resolveResource(@NonNull context: Context, resId: Int): Int {
        val outValue = TypedValue()
        context.theme.resolveAttribute(resId, outValue, true)
        return outValue.resourceId
    }
}